src/main/java/com/cloudbees/AWSCredentialsUtils.java

```diff
@@ -0,0 +1,60 @@
+package com.cloudbees.jenkins.plugins.awscredentials;
+
+import com.amazonaws.ClientConfiguration;
+import com.amazonaws.auth.AWSCredentialsProvider;
+import com.amazonaws.regions.DefaultAwsRegionProviderChain;
+import com.amazonaws.regions.Regions;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenService;
+import com.amazonaws.services.securitytoken.AWSSecurityTokenServiceClientBuilder;
+
+import hudson.ProxyConfiguration;
+import jenkins.model.Jenkins;
+
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+final class AWSCredentialsUtils {
+
+    private static final Logger LOGGER = Logger.getLogger(AWSCredentialsUtils.class.getName());
+
+    private AWSCredentialsUtils() {}
+
+    static AWSSecurityTokenService buildStsClient(AWSCredentialsProvider provider) {
+        // Check for available region from the SDK, otherwise specify default
+        String clientRegion = null;
+        DefaultAwsRegionProviderChain sdkRegionLookup = new DefaultAwsRegionProviderChain();
+        try {
+            clientRegion = sdkRegionLookup.getRegion();
+        }
+        catch(com.amazonaws.SdkClientException e) {
+            LOGGER.log(Level.WARNING,"Could not find default region using SDK lookup.", e);
```

Comment:
Based on the comment on the `AwsRegionProviderChain#getRegion`, this exception shouldn't be thrown. 
It seems to be a bug in the library that shouldn't be in 2.0 (again, based on library comment).

It might be good not to log this as it is, to not let users think there is a bug in this plugin.

```suggestion
            LOGGER.log(Level.INFO, "SDK Region lookup throw an exception incorrectly. We will use the default region.", e);
```

Reply:
This is from the existing implementation. https://github.com/jenkinsci/aws-credentials-plugin/blob/2ae45df5330d9e64a574979736def83fa8dece0f/src/main/java/com/cloudbees/jenkins/plugins/awscredentials/AWSCredentialsImpl.java#L141

I can change it if you want, I just didn't want to alter the existing behavior.

Reply:
Also I'm not sure what you mean by 2.0. This is using v1.11.341 of the SDK.

Reply:
> Also I'm not sure what you mean by 2.0. This is using v1.11.341 of the SDK.

from the comment on the source code of the library. The exception shouldn't be thrown but 🤷🏻‍♂️